### PR TITLE
Ansible Tower provisoning callback

### PR DESCRIPTION
**Pull request by dLobatog**
Service to pull and run provisioning scripts from Ansible tower
post-installation. The snippet is compatible only with systemd-enabled
servers for the moment

props & shoutouts to @wzzrd

### CODE REVIEW

**Comment by wzzrd** [comment]
Yay!

**Comment by ekohl** [comment]
Why such a complicated solution to do one curl call? We do the foreman post-installation HTTP call during provisioning (like [here](https://github.com/theforeman/community-templates/blob/5ca7947f02a4989fd21ecd4f9654df705c1577b3/provisioning_templates/provision/kickstart_default.erb#L198)).

**Comment by dLobatog** [comment]
@ekohl I guess the benefit of this is that you would be able to repeat this any time by just restarting the service

**Comment by ekohl** [comment]
To me it doesn't feel like a service. You could also write a ```/usr/local/bin/ansible-tower-callback``` and execute that during provisioning. It'd have the same benefits and not be limited to systemd. While most Linux distro's are switching over, the BSDs we support won't.

**Comment by wzzrd** [comment]
So the point of doing it this way is that the callback to Tower needs to happen *after* the reboot after kickstarting. 

The callback is going return immediately and will trigger an Ansible run from Tower, so we don't want to reboot right after the cURL call returns, because that will likely make the Ansible run from Tower fail. The host will probably either unreachable during reboot or shutting down after kickstarting.

Reason I used a service for this was that services have been used in this way before, and it was an easy solution when I really needed it. 

I am sure we can improve upon this and rewrite the snippet into something that is more portable though.

**Comment by ekohl** [comment]
Ah, so it's more than "just registration". I wasn't aware of the callback. In that case it does make sense.

File: provisioning_templates/provision/kickstart_default.erb [hunk 1]
```diff
@@ -191,6 +191,17 @@ yum -t -y remove kernel-uek*
 sed -e 's/DEFAULTKERNEL=kernel-uek/DEFAULTKERNEL=kernel/g' -i /etc/sysconfig/kernel
 <% end -%>
 
+# This is RHEL7 specific code, as it uses systemctl, replace with
+# chkconfig or an @boot cron if you run RHEL5 or RHEL6
+<% if realm_compatible && @host.param_true?('ansible_tower_provisioning') %>
+cat > /etc/systemd/system/ansible-callback.service << EOF
```
Comment (mmoll) [inline provisioning_templates/provision/kickstart_default.erb]: can `save_to_file` be used here?

Comment (dLobatog) [reply provisioning_templates/provision/kickstart_default.erb]: indeed

**Review by dLobatog — COMMENTED** [review]
Updated to use `save_to_file`, thanks for the tip @mmoll

**Comment by mmoll** [comment]
this needs to use `host_param_true?` now.

**Comment by mmoll** [comment]
merged after updating to new macros. gracis @dLobatog & bedankt @wzzrd!

File: provisioning_templates/provision/kickstart_default.erb [hunk 2]
```diff
@@ -192,6 +192,15 @@ yum -t -y remove kernel-uek*
 sed -e 's/DEFAULTKERNEL=kernel-uek/DEFAULTKERNEL=kernel/g' -i /etc/sysconfig/kernel
 <% end -%>
 
+# This is RHEL7 specific code, as it uses systemctl, replace with
```
Comment (ekohl) [inline provisioning_templates/provision/kickstart_default.erb:195]: Nit that's a bit late: The comment should be inside the if statement so you won't end up with just a comment when it's false. I've submitted https://github.com/theforeman/community-templates/pull/410 to fix this.

**Pull request merged**